scipy/spatial/transform/_rotation.pyx

```diff
@@ -1273,6 +1386,186 @@ cdef class Rotation:
         else:
             return cls(quat, normalize=False, copy=False)
 
+    @cython.embedsignature(True)
+    @classmethod
+    def from_davenport(cls, axes, angles, extrinsic, degrees=False):
+        """Initialize from Davenport angles.
+
+        Rotations in 3-D can be represented by a sequence of 3
+        rotations around a sequence of axes.
+
+        The three rotations can either be in a global frame of reference
+        (extrinsic) or in a body centred frame of reference (intrinsic), which
+        is attached to, and moves with, the object under rotation [1]_.
+
+        For Euler angles, we have the following conditions:
+
+            - ``dot(axis1, axis2) = 0``
+            - ``dot(axis2, axis3) = 0``
+            - ``dot(axis1, axis3) = -1``, ``0`` or ``1``
+
+        For Davenport angles, the third relation is relaxed [2]_:
+
+            - ``dot(axis1, axis3)`` can be any value between -1 and 1.
+
+        Parameters
+        ----------
+        axes : array_like
+            Specifies sequence of axes for rotations. Must be an array_like of
+            shape (3, ) or (3, [1 or 2 or 3]), where each axes[i, :] is the ith
```

Comment:
Do you mean arrays like:

- [1, -1, 0.5] - single axis
- [[1, -1, 0.5], [-1, 1, 0]] - two axes

It looks like the first dimension can vary, not second. Am I correct?

Reply:
Correct, basically, there are four possibilities:

1. `[ax, ay, az]` which has shape `(3, )`
2. `[[ax, ay, az]]` which has shape `(3, 1)`
2. `[[ax, ay, az], [bx, by, bz]]` which has shape `(3, 2)`
2. `[[ax, ay, az], [bx, by, bz], [cx, cy, cz]]` which has shape `(3, 3)`

Reply:
I think you are mixing up 0 and 1 axis. In your third example the shape will be (2, 3) - 2 rows and 3 columns.

This caused the confusion for me.

Reply:
Oops you're right, fixing it right away